### PR TITLE
remove!: Removes old Id format, only `idp_id` can be used as id for `mongodbatlas_federated_settings_identity_provider`

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Only `idp_id`can be used as id for `mongodbatlas_federated_settings_identity_provider`.

As in the doc:
> WARNING: Starting from terraform provider version 1.16.0, to import the resource a 24-hexadecimal digit string that identifies the IdP (idp_id) will have to be used instead of okta_idp_id

The PR keeps 2 SDK versions as it'll be needed when updating to the latest SDK.   `.golangci.ym` has been changed temporarily because both SDKs are the same in this PR, but that linter config check should be reverted when the new SDK is updated.

Link to any related issue(s): CLOUDP-244544

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: website/docs/guides/1.16.0-upgrade-guide.html.markdown [hunk 1]
```diff
@@ -12,11 +12,12 @@ The Terraform MongoDB Atlas Provider version 1.16.0 has a number of new and exci
...
 
 **Deprecations and Removals:**
 
-- The HashiCorp Terraform Version Compatibility Matrix has been published for the Terraform MongoDB Atlas Provider. See full details at https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs#hashicorp-terraform-version-compatibility-matrix
+- Format of IdP Id that uniquely identifies the identity provider when importing [`mongodbatlas_federated_settings_identity_provider`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/federated_settings_identity_provider) resource and [`mongodbatlas_federated_settings_identity_provider`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/data-sources/federated_settings_identity_provider) data source only accepts the new IdP Id format that is a 24-hex characters long string.
+More info can be found in the [previous guide](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/1.15.0-upgrade-guide).
```
Comment (oarbusi) [inline website/docs/guides/1.16.0-upgrade-guide.html.markdown]:
```suggestion
More info can be found in the [previous upgrade guide](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/1.15.0-upgrade-guide).
```

**Review by oarbusi — APPROVED** [review]
LGTM!

**Review by AgustinBettati — APPROVED** [review]
LGTM

**Review by maastha — APPROVED** [review]

File: website/docs/guides/1.16.0-upgrade-guide.html.markdown [hunk 2]
```diff
@@ -12,11 +12,12 @@ The Terraform MongoDB Atlas Provider version 1.16.0 has a number of new and exci
 
 **New Resources, Data Sources, and Features:**
 
-
+- The HashiCorp Terraform Version Compatibility Matrix has been published for the Terraform MongoDB Atlas Provider. See full details at https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs#hashicorp-terraform-version-compatibility-matrix
```
Comment (Zuhairahmed) [inline website/docs/guides/1.16.0-upgrade-guide.html.markdown]:
```suggestion
- The HashiCorp Terraform Version Compatibility Matrix has been published for the Terraform MongoDB Atlas Provider. See full details at https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs#hashicorp-terraform-versionhttpswwwterraformiodownloadshtml-compatibility-matrix 
```

**Review by Zuhairahmed — APPROVED** [review]
minor comment, otherwise LGTM

**Pull request merged**